dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/ajax/ContentletAjax.java

```diff
@@ -851,209 +853,219 @@ else if( fieldbcontentname.startsWith("date") ){
-			Identifier ident=APILocator.getIdentifierAPI().find(con);
-			if(!con.isLive()) {
-    			if(UtilMethods.isSet(ident.getSysExpireDate()) && ident.getSysExpireDate().before(new Date()))
-    			    expiredInodes.add(con.getInode()); // it is unpublished and can't be manualy published
-			}
 
-			Map<String, String> searchResult = new HashMap<String, String>();
-			Structure s = CacheLocator.getContentTypeCache().getStructureByInode(con.getStructureInode());
+			Map<String, String> searchResult = null;
 
-			for (String fieldContentlet : fieldsMapping.keySet()) {
-				String fieldValue = null;
-				if(con.getMap() != null && con.getMap().get(fieldContentlet) !=null){
-					fieldValue =(con.getMap().get(fieldContentlet)).toString();
+			try {
+				con = hits.get(i);
+				Identifier ident = APILocator.getIdentifierAPI().find(con);
+				if (!con.isLive()) {
+					if (UtilMethods.isSet(ident.getSysExpireDate()) && ident.getSysExpireDate().before(new Date()))
+						expiredInodes.add(con.getInode()); // it is unpublished and can't be manualy published
 				}
-				Field field = (Field) fieldsMapping.get(fieldContentlet);
-				if (UtilMethods.isSet(fieldValue) && field.getFieldType().equals(Field.FieldType.DATE.toString()) ||
-						UtilMethods.isSet(fieldValue) && field.getFieldType().equals(Field.FieldType.TIME.toString()) ||
-						UtilMethods.isSet(fieldValue) && field.getFieldType().equals(Field.FieldType.DATE_TIME.toString())) {
-					try {
-						Date date = DateUtil.convertDate(fieldValue, new String[] { "yyyy-MM-dd HH:mm:ss", "E MMM dd HH:mm:ss z yyyy" });
-						if (field.getFieldType().equals(Field.FieldType.DATE.toString()))
-							fieldValue = UtilMethods.dateToHTMLDate(date);
-						if (field.getFieldType().equals(Field.FieldType.TIME.toString()))
-							fieldValue = UtilMethods.dateToHTMLTime(date);
-						if (field.getFieldType().equals(Field.FieldType.DATE_TIME.toString()))
-							fieldValue = UtilMethods.dateToHTMLDate(date) + " " + UtilMethods.dateToHTMLTime(date);
-					} catch (java.text.ParseException e) {
-						Logger.error(ContentletAjax.class, e.getMessage(), e);
-						throw new DotRuntimeException(e.getMessage(), e);
+
+				searchResult = new HashMap<String, String>();
+				Structure s = CacheLocator.getContentTypeCache().getStructureByInode(con.getStructureInode());
+
+				for (String fieldContentlet : fieldsMapping.keySet()) {
+					String fieldValue = null;
+					if (con.getMap() != null && con.getMap().get(fieldContentlet) != null) {
+						fieldValue = (con.getMap().get(fieldContentlet)).toString();
+					}
+					Field field = (Field) fieldsMapping.get(fieldContentlet);
+					if (UtilMethods.isSet(fieldValue) && field.getFieldType().equals(Field.FieldType.DATE.toString()) ||
+							UtilMethods.isSet(fieldValue) && field.getFieldType().equals(Field.FieldType.TIME.toString()) ||
+							UtilMethods.isSet(fieldValue) && field.getFieldType().equals(Field.FieldType.DATE_TIME.toString())) {
+						try {

```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS134)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested try block into a separate method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)

Reply:
Legacy code, do not wanna touch